R/shave.R

```diff
@@ -1,8 +1,27 @@
-shave.matlist <- function(mat_list, nb_cols)
-  mapply(function(m, nbcomp) m[, 1:nbcomp, drop = FALSE],
-         mat_list, nb_cols,
-         SIMPLIFY = FALSE
-  )
-
-shave.veclist <- function(vec_list, nb_elts)
-  mapply(function(m, nbcomp) m[1:nbcomp], vec_list, nb_elts, SIMPLIFY = FALSE)
+shave <- function(m, nbcomp) {
+    UseMethod("shave")
+}
```

Comment:
I didn't know about this, this is great!

Reply:
It uses the S3 object oriented methods to work, so that if there is a method for that class it will be used. This is like the plot.bootstrap which only works on boostrap class.

Reply:
Please update the Namespace with devtools::document()